### ISSUE TITLE
cache should check local filesystem before downloading

### ISSUE THREAD

**Issue by NelsonMinar** [openaddresses/machine#39]
`cache.py` seems to have no code to check if a download should occur. It'd be nice if it checked if the source changed first before doing the network traffic. In many cases we can't cache; either it's a run on a new directory or it's a web service that doesn't support checks. But when debugging we could save a lot of time looking on disk. Could also turn into a system to avoid regenerating outputs if the input didn't change.

Specific proposal: add code to `download()` [here](https://github.com/openaddresses/machine/blob/26f24fa69300bd8c83db1512b0421ac1e099c7e7/openaddr/cache.py#L107-L143) to do a check against the server first. Implementing the full HTTP caching protocol is too much work, but maybe we can get by with a check on content length and date.

Looking at that code now it looks like in some circumstances the file won't be redownloaded if it exists. I haven't seen that work in practice and, even if it does, we should be smarter about checking for cache expiration.

**Comment by NelsonMinar** [comment]
Does anyone else want this feature? I haven't really needed it, on occasion I've worked around it by altering the download URL in a local source file. If no one speaks up I'll close this issue without doing any work.

I think we do want to revisit this question in the larger context of repeated regular machine runs. It'd be nice to detect if the source has not changed, avoid processing a file if the result will be the same as last time we ran it.
